### ISSUE TITLE
Optional field-specific validation

### ISSUE THREAD

**Issue by ajmnz** [prisma/prisma#8357]
## Problem

Nowadays, almost every app that deals with user data in general has columns to store email addresses, URLs, etc. I thought it would be a really nice addition to have Prisma-level field validation that would make sure that certain data has a specific format and throw an error if otherwise.

It is obvious that validation has to be done before inserting anything into a database, but having this optional extra layer of protection _in case something happens_, would save some headaches and open a lot more doors for Prisma.

## Suggested solution

For example, it would be really nice if we could do something like the following directly in our models

```prisma
model User {
  id    Int    @id @default(autoincrement())
  name  String
  email String @unique @isEmail // <--
  Bio   Bio
}

model Bio {
  id      Int     @id @default(autoincrement())
  content String?
  url     String  @isUrl // <--
}

```

Or even custom regex validation

```prisma
model ShippingAddress {
  id      Int    @id @default(autoincrement())
  name    String
  zipCode Int    @isRegex(/^\d{5}(?:[- ]?\d{4})?$/)
}
```

## Alternatives

None that I can think of, rather than regular validation before inserting anything in a database.

## Additional context

As a side note, think the possibilities this would open for autocompletion if [this proposal](https://github.com/microsoft/TypeScript/issues/41160) goes forward and gets implemented into TypeScript.

**Comment by sandbox-apps** [comment]
I think I would also need this on our projects. +1 to this. Really love this idea that you can do regex and some predefaulted regex like (isUrl, isEmail) we can get inspiration to this package https://github.com/typestack/class-validator for almost everything like @IsEmpty(), @IsNotEmpty(), @IsPositive(), @IsNegative(), @IsCreditCard(), @IsHexColor(), @IsLatLong(), @IsPhoneNumber(region: string)..

It would be great if it will be persisted to db level as constraint check. For strict security...

**Comment by pantharshit00** [comment]
This issue is duplicate of https://github.com/prisma/prisma/issues/3102 + https://github.com/prisma/prisma/issues/3528

Related: https://github.com/prisma/prisma/issues/7209
